File: app/android/app/src/main/java/com/raunderz/icps/WifiScannerPackage.java
package com.raunderz.icps;

import com.facebook.react.BaseReactPackage;
import com.facebook.react.bridge.ModuleSpec;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import java.util.Arrays;
import java.util.List;

public class WifiScannerPackage extends BaseReactPackage {
  @Override
  public List<ModuleSpec> getNativeModules(ReactApplicationContext reactContext) {
    return Arrays.<ModuleSpec>asList(
      new ModuleSpec(WifiScannerModule.class, () -> new WifiScannerModule(reactContext))
    );
  }
}

File: kt_app/app/src/main/java/com/ibnips/kotlinapp/bridge/WifiScannerPackage.kt
package com.ibnips.kotlinapp.bridge

import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager

/**
 * React Native package for the Wi-Fi Scanner module.
 * 
 * Registers the [WifiScannerModule] to be accessible from the JavaScript layer.
 */
class WifiScannerPackage : ReactPackage {

    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
        return listOf(WifiScannerModule(reactContext))
    }

    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
        return emptyList()
    }
}
